client/src/main/scala/com/all4journey/webapp/util/Fragments.scala

```diff
@@ -0,0 +1,151 @@
+import com.wix.accord.{Success => ValidationSuccess}
+import prickle.Unpickle
+
+import scala.scalajs.js
+import scala.scalajs.js.annotation.JSExport
+import scala.util.Success
+import scalatags.JsDom.all._
+import com.all4journey.shared.domain._
+
+// $COVERAGE-OFF$
+trait NavPills {
+  @JSExport
+  def getNavPills(active:String) = ul(cls := "nav nav-pills")(
+    li(id := "personalInfoLink", role := "presentation", if(active.equalsIgnoreCase("personalInfoLink")) cls := "active" else "")(
+      a(href := "/multiformProfile/personal")("Personal Info")
+    ),
+    li(id := "placesLink", role := "presentation", if(active.equalsIgnoreCase("placesLink")) cls := "active" else "")(
+      a(href := "/multiformProfile/places")("Your Places")
+    ),
+    li(id := "passwordChangeLink", role := "presentation", if(active.equalsIgnoreCase("passwordChangeLink")) cls := "active" else "")(
+      a(href := "/multiformProfile/password")("Change Password")
+    )
+  )
+}
+
+trait AddressForm extends AddressTypePickler {
+
+  @JSExport
+  val emptyAddress = Address("0", "0", None, State("NONE", "Choose a state"), "", HomeAddressType, "Home")
+
+  @JSExport
+  def getAddressForm(address: Option[Address]) = div(id := "addressFieldsDiv", "addressUuid".attr := address.getOrElse(emptyAddress).id)(
+
+    div(id := "streetAddressDiv", cls := "form-group")(
+      label(cls := "col-lg-3 control-label")("Street Address:"),
+      div(cls := "col-lg-8")(
+        div(id := "streetAddressHelpBlock"),
+        input(id := "streetAddress", name := "streetAddress", cls := "form-control", `type` := "text", value := address.getOrElse(emptyAddress).street.getOrElse(""))
+      )
+    ),
+    div(id := "stateDiv", cls := "form-group")(
+      label(cls := "col-lg-3 control-label")("State:"),
+      div(cls := "col-lg-8")(
+        div(id := "stateHelpBlock"),
+        div(cls := "ui-select")(
+          select(id := "userState", name := "userState", cls := "form-control partOfStateList")(
+            option(value := address.getOrElse(emptyAddress).state.id, selected := "selected")(address.getOrElse(emptyAddress).state.description)
+          )
+        )
+      )
+    ),
+    div(id := "zipCodeDiv", cls := "form-group")(
+      label(cls := "col-lg-3 control-label")("Zip code:"),
+      div(cls := "col-lg-8")(
+        div(id := "zipCodeHelpBlock"),
+        input(id := "zipCode", name := "zipCode", cls := "form-control partOfZipCodeList", `type` := "text", value := address.getOrElse(emptyAddress).zipCode)
+      )
+    )
+  )
+
+  def setUiViolationPrompts(violations: Set[Violation]) = {
+    if (!violations.isEmpty) {
+      //                      val content = dom.document.getElementById("content")
+      //                      for (v <- failureList)
+      //                        content.appendChild(p(v.description).render)
+      violations.foreach(violation =>
+        if (violation.description.getOrElse("").contains("street")) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Consider using case matching instead of else if blocks](https://www.codacy.com/app/rjkernick/all4journey/file/2503283144/issues/source?bid=3246460&fileBranchId=3273764#l72)